### PR TITLE
fix for not re-enabling preview on leaving security-sensitive pages in marketplace

### PR DESCRIPTION
**Pull request by ElderOrb**
All pages that trigger the commerce specific disabled preview screen in the marketplace (places where user enters their wallet pin) do not dismiss the disabled preview screen when credentials are not entered, but rather the user selects the "Cancel" button or the tablet "x" button.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7476/app_12491_PR.apk

**Comment by ElderOrb** [comment]
*** Test plan *** 

1) Enable preview
2) Switch to VR
3) Navigate to the Marketplace using the Tablet
4) Select the "My Purchases" link to enable the commerce specific disabled preview screen
5) Select the "Cancel" button on the wallet PIN page
6) Ensure preview re-enabled 
7) Repeat steps 1-6 with clicking 'X' button (go to homescreen) instead of 'Cancel' button on step 5. 

Please also check other possible places in marketplaces where security-sensitive screens might be presented.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7476/HighFidelity-Beta-PR12491-f9de255fa7d294c08452591b2c81fa1471d87bca.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7476/HighFidelity-Beta-PR12491-f9de255fa7d294c08452591b2c81fa1471d87bca.dmg

**Comment by qlimnavarro** [comment]
Currently Testing.

**Comment by qlimnavarro** [comment]
Fail
When in the wallet app and user enters Change password page, security-sensitive screen is displayed. When the user selects cancel, the security-sensitive screen will not go away until the wallet app is closed. Comparing this to the change security picture button which will re-enable review when the cancel button is selected is it not consistent. Both will re-enable preview when X button on tablet is selected.
Steps to repro:
1) With HMD open wallet
2) Select Security
3) Select Change Password
4) Note Preview is disabled on desktop
5) Select cancel
6) Note on desktop preview is still disabled
7) Close Wallet app
8) Preview is re-enabled
9) Compare to change security pic functionality

All other passed scenarios:
1) Wallet password through Wallet app
2) Wallet password through my purchases
3) Wallet password through buying an item
4) Wallet password when changing security picture.
All of the above were tested by hitting cancel button or X on tablet.

Video and Log attached (Might be hard to see in video, by following step in video, HMD user can see the defect)
[PR12491 Files.zip](https://github.com/highfidelity/hifi/files/1764527/PR12491.Files.zip)
@conklin94122

**Comment by conklin94122** [comment]
@themelissabrown Needs milestone

**Comment by ElderOrb** [comment]
Ah, good catch! Fixed 'change password' case, please re-check

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7608/app_12491_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7608/HighFidelity-Beta-PR12491-bee3f37fa211bb07e52fb8eee473ef952add279b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7608/HighFidelity-Beta-PR12491-bee3f37fa211bb07e52fb8eee473ef952add279b.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass - @conklin94122

**Comment by conklin94122** [comment]
@AndrewMeadows @zfox23 Needs dev review

File: interface/resources/qml/hifi/commerce/wallet/Wallet.qml [hunk 1]
```diff
@@ -222,6 +222,8 @@ Rectangle {
                     } else {
                         sendToScript(msg);
                     }
+                } else if(msg.method === 'maybeEnableHmdPreview') {
```
Comment (zfox23) [inline interface/resources/qml/hifi/commerce/wallet/Wallet.qml]: Will this work if `passphraseChange.visible === true` ?

I'm confused by this logic.

Comment (ElderOrb) [reply interface/resources/qml/hifi/commerce/wallet/Wallet.qml]: If `passphraseChange.visible === true` preview shouldn't be enabled.

**Comment by ElderOrb** [comment]
fixed styling

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7752/app_12491_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7752/HighFidelity-Beta-PR12491-53e1a0cae5fa41ba691e65caace1058867f2da91.dmg

**Comment by changxc** [comment]
thanks for dev and QA approve! can merge to RC 66 :)

**Comment by zfox23** [comment]
@changxc - will merge once gustavo finishes

**Comment by zfox23** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7792/app_12491_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7792/HighFidelity-Beta-PR12491-53e1a0cae5fa41ba691e65caace1058867f2da91.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7792/HighFidelity-Beta-PR12491-53e1a0cae5fa41ba691e65caace1058867f2da91.dmg

**Pull request merged**